src/bridges/aave/lending/AaveLendingBridgeConfigurator.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import {Ownable} from '@openzeppelin/contracts/access/Ownable.sol';
+import {IAaveLendingBridgeConfigurator} from './interfaces/IAaveLendingBridgeConfigurator.sol';
+import {IAaveLendingBridge} from './interfaces/IAaveLendingBridge.sol';
+import {ILendingPool} from './../imports/interfaces/ILendingPool.sol';
+import {IPool} from './../imports/interfaces/IPool.sol';
+
+/**
+ * @notice AaveLendingBridgeConfigurator implementation that is used to add new listings to the AaveLendingBridge
+ * permissionlessly deposit and withdraw funds into the listed reserves. Configurator cannot remove nor update listings
+ * @author Lasse Herskind
+ */
+contract AaveLendingBridgeConfigurator is IAaveLendingBridgeConfigurator, Ownable {
+    function addNewPool(
+        address lendingBridge,
+        address underlyingAsset,
+        address aTokenAddress
+    ) public override(IAaveLendingBridgeConfigurator) onlyOwner {
+        IAaveLendingBridge(lendingBridge).setUnderlyingToZkAToken(underlyingAsset, aTokenAddress);
+    }
+
+    function addPoolFromV2(address lendingBridge, address underlyingAsset)
+        external
+        override(IAaveLendingBridgeConfigurator)
+    {
+        IAaveLendingBridge bridge = IAaveLendingBridge(lendingBridge);
+        ILendingPool pool = ILendingPool(bridge.ADDRESSES_PROVIDER().getLendingPool());
+
+        address aTokenAddress = pool.getReserveData(underlyingAsset).aTokenAddress;
+
+        addNewPool(lendingBridge, underlyingAsset, aTokenAddress);
+    }
+
+    function addPoolFromV3(address lendingBridge, address underlyingAsset)
+        external
+        override(IAaveLendingBridgeConfigurator)
+    {
+        IAaveLendingBridge bridge = IAaveLendingBridge(lendingBridge);
+        IPool pool = IPool(bridge.ADDRESSES_PROVIDER().getLendingPool());
+
+        address aTokenAddress = pool.getReserveData(underlyingAsset).aTokenAddress;
+
+        addNewPool(lendingBridge, underlyingAsset, aTokenAddress);
+    }
+
+    function claimLiquidityRewards(
+        address lendingBridge,
+        address incentivesController,
+        address[] calldata assets,
+        address beneficiary
+    ) external override(IAaveLendingBridgeConfigurator) onlyOwner returns (uint256) {
```

Comment:
Wouldn't it be better if the rewards were automatically distributed/made claimable by the zkToken holders? Would be more permissionless, no?

In Liquity bridges, I am automatically dumping rewards token for the underlying to make claiming more gas efficient.

Reply:
We are not claiming and swapping because the massive gas-costs that is incurred by this (can outweigh the rewards if done often). The gas-costs for Aave and stkAave is very high, (transfer can be greater than 200K gas).